apps/genpkey.c

```diff
@@ -294,14 +294,25 @@ int genpkey_main(int argc, char **argv)
 
  end:
     sk_OPENSSL_STRING_free(keyopt);
-    if (ret != 0)
+    if (ret != 0) {
         ERR_print_errors(bio_err);
+    } else {
+        if (mem_outpubkey != NULL) {
+            rv = mem_bio_to_file(mem_outpubkey, outpubkeyfile, outformat, private);
+            if (!rv)
+                BIO_puts(bio_err, "Error writing to outpubkey\n");
```

Comment:
Also report the filename here:
```suggestion
                BIO_puts(bio_err, "Error writing to pubkey output file: %s\n", outpubkeyfile);
```
And it would be useful to know why the write failed (permission issue, space issue, ...), so either print the error stack or the "strerror(errno)" value.

Reply:
Updated, now including the filename and strerror(errno)